src/app/loadout-drawer/loadout-item-conversion.ts

```diff
@@ -44,15 +45,44 @@ export function getItemsFromLoadoutItems(
       const modsForBucket =
         loadoutItem.equip && modsByBucket ? modsByBucket[item.bucket.hash] ?? [] : [];
 
-      let overrides = loadoutItem.socketOverrides;
+      let overrides = loadoutItem.socketOverrides || {};
 
+      // Apply armour mods to the socket overrides
       for (const modHash of modsForBucket) {
         const socket = item.sockets?.allSockets.find((s) => plugFitsIntoSocket(s, modHash));
         if (socket) {
           overrides = { ...overrides, [socket?.socketIndex]: modHash };
         }
       }
 
+      // Hydrate the empty sockets for fragments as overrides. This ensures that the stats that
```

Comment:
What does this mean, to hydrate them?

Reply:
I meant that we replace the inactive fragments with empty sockets so that when the stat calculations are run that actual stat state of the loadout would be correct.

The crux of this whole PR was to "fix" the issue with inactive fragments applying stats to items but having a truer representation of active fragments.